### PR TITLE
Adding screen reader optimized mode to slickgrid

### PR DESCRIPTION
**Pull request by aasimkhan30**
Issue:
https://github.com/microsoft/azuredatastudio/issues/20784
https://github.com/microsoft/azuredatastudio/issues/20780

### CODE REVIEW

File: src/sql/base/browser/ui/accessibility/accessibilityProvider.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+export interface IAccessibilityProvider {
```
[the remark this answers is not in the document]
Comment (aasimkhan30) [reply src/sql/base/browser/ui/accessibility/accessibilityProvider.ts:6]: I didn't want to repeat the logic everywhere. That's why I decided to pass the service instance

[the remark this answers is not in the document]
Comment (alanrenmsft) [reply src/sql/base/browser/ui/accessibility/accessibilityProvider.ts:6]: what logic? isn't it as simple as accessibilityService.screenreaderenabled?

[the remark this answers is not in the document]
Comment (alanrenmsft) [reply src/sql/base/browser/ui/accessibility/accessibilityProvider.ts:6]: after the offline chat, I think it makes sense, we always want this to be the same as the setting.

**Comment by alanrenmsft** [comment]
@aasimkhan30 don't forget to link the issues.

**Pull request merged**